Packs/MandiantAdvantageAttackSurfaceManagement/Integrations/AttackSurfaceManagement/AttackSurfaceManagement.py

```diff
@@ -0,0 +1,467 @@
+
+        return response['result']
+
+    def get_notes(self,
+                  resource_type: str,
+                  resource_id: str) -> dict:
+        endpoint = f'notes/{resource_type}/{resource_id}'
+
+        response = self._http_request('GET', endpoint,
+                                     headers={'PROJECT_ID': str(self.project_id)})
+
+        if not isinstance(response, dict) or not response.get('success'):
+            raise DemistoException('The ASM API was unable to return notes for'
+                                   f'{resource_type} {resource_id} in project {self.project_id}')
+
+        return response.get('result') or []
+
+
+''' HELPER FUNCTIONS '''
+
+def helper_create_incident(issue: dict, project_id: int) -> dict:
+    severity_map = {1: 4, 2: 3, 3: 2, 4: 1, 5: 0.5}
+
+    raw_json = issue
+    raw_json['mirror_direction'] = 'Both'
+    raw_json['mirror_instance'] = demisto.integrationInstance()
+
+    issue = {
+        'name': issue['summary']['pretty_name'],
+        'details': json.dumps(issue, indent=4),
+        'occurred': parser.parse(issue['first_seen']).isoformat(),
+        'severity': severity_map.get(issue['summary']['severity']),
+        'rawJSON': json.dumps(raw_json),
+        'dbotMirrorId': issue.get('id', issue.get('uid')),
+        'CustomFields': {
+            'masmcollection': issue['collection'],
+            'masmproject': str(project_id),
+            'masmconfirmed': issue['summary']['confidence'] == 'confirmed',
+            'masmentityname': issue['entity_name'],
+            'masmcategory': issue['summary']['category']
+        }
+    }
+
+    return issue
+
+def fetch_incident_helper(client: Client, last_run: dict) -> typing.Tuple[dict, list]:
+    last_start_time: datetime
+
+    params = demisto.params()
+
+    if not client.project_id:
+        raise DemistoException('Must configure a Project ID to fetch incidents from Mandiant ASM')
+
+    if last_run and 'start_time' in last_run and last_run['start_time']:
+        last_start_time = parser.parse(last_run.get('start_time'))
+    else:
+        parsed_time = dateparser.parse(params.get("first_fetch"), settings={'TIMEZONE': 'UTC', 'RETURN_AS_TIMEZONE_AWARE': True})
+        if not parsed_time:
+            parsed_time = datetime.now()
+        last_start_time = parsed_time
+
+    issues = client.get_issues_since_time(last_start_time)
+    most_recent_update = None
+
+    parsed_issues = []
+
+    for issue in issues:
+        incident_details = client.get_issue_details(issue["id"])
+        new_incident = helper_create_incident(incident_details, client.project_id)
+        new_incident['dbotMirrorLastSync'] = datetime.fromtimestamp(0).strftime("%Y-%m-%dT%H:%M:%SZ")
+        new_incident['lastupdatetime'] = new_incident['dbotMirrorLastSync']
+
+        parsed_issues.append(new_incident)
+        most_recent_update = issue['last_seen']
+
+        if len(parsed_issues) >= client.limit:
+            break
```

Comment:
maybe you should filter those **after** the **filter_incidents_by_duplicates_and_limit** function.

Reply:
@chrishultin what do you think?

Reply:
During testing, I found that removing that limit resulted in timeouts, as the client would continue fetching new indicators that would then be discarded as part of the filter-and-limit call.